cray_product_catalog/query.py

```diff
@@ -174,6 +174,32 @@ def get_product(self, name, version=None):
         return matching_products[0]
 
 
+def load_config_map_data(name, configmaps):
+    """Parse list_namespaced_config_map output and get array of InstalledProductVersion objects.
+
+    Args:
+        configmaps (V1ConfigMapList): list of ConfigMap objects.
+
+    Returns:
+        An array of InstalledProductVersion objects.
+    """
+    config_map_data = {}
+    for cm in configmaps:
+        if not cm.metadata.name.startswith(name):
+            continue
+        for product_name, product_versions in cm.data.items():
+            for product_version, product_version_data in safe_load(product_versions).items():
+                cm_key = product_name + ':' + product_version
```

Comment:
Instead of combining these as a string separated by a colon, why not just store them in a nested dictionary which has the product name as its first level of keys and the product version as its second level of keys. Then we don't need to do any string manipulation that could be thrown off by a colon existing in the product name or version.

Reply:
I have not tested this at all, but I was thinking something like this for the whole function:

```
    config_map_data = {}
    for cm in configmaps:
        if not cm.metadata.name.startswith(name):
            continue
        for product_name, product_versions in cm.data.items():
            if product_name not in config_map_data:
                config_map_data[product_name] = {}
            for product_version, product_version_data in safe_load(product_versions).items():
                if product_version in config_map_data[product_name]:
                    config_map_data[product_name][product_version] = merge_dict(
                        config_map_data[product_name][product_version],
                        product_version_data
                    ) 
                else:
                    config_map_data[product_name][product_version] = product_version_data
    return [
        InstalledProductVersion(product_name, product_version, product_version_data)
        for product_name, product_versions in config_map_data.items()
        for product_version, product_version_data in product_versions.items()
    ]
```

Reply:
Modified as mentioned.